### ISSUE TITLE
Double division by batch size

### ISSUE THREAD

**Issue by bkj** [williamleif/GraphSAGE#12]
In the unsupervised model, you do this operation twice:
```
self.loss = self.loss / tf.cast(self.batch_size, tf.float32)
```
on the following lines:

 - https://github.com/williamleif/GraphSAGE/blob/master/graphsage/models.py#L378
 - https://github.com/williamleif/GraphSAGE/blob/master/graphsage/models.py#L391

Is the repetition intentional?  Or is this a bug?

**Comment by williamleif** [comment]
This is unintentional and means that our learning rate is effectively scaled (unnecessarily) by the batch size. I've removed it and will update the paper to reflect the unscaled learning rate values (camera-ready will be submitted to arxiv today, or very early tomorrow).

I will also add an acknowledgement to you in the paper. Rex and I really appreciate your comments, questions, and careful reading of our code. 

Also, note that the hinge_loss turned out to be much better than the xent loss in follow-up work. We are keeping the xent loss (and other hyperparameter) settings as they were in the original NIPS submission, because they were designed to give a fair comparison to node2vec etc., but I recommend the hinge_loss if you want to do link prediction-like tasks.

**Comment by bkj** [comment]
OK makes sense -- figured it was a bug.  

Thanks for the acknowledgement -- I really appreciate you guys releasing the code in the first place!

Good to know about the hinge loss -- I'll implement that in my fork.

One more question -- do you have the parameters for the unsupervised reddit benchmarks written down somewhere?
